integration/spark-common/src/main/scala/org/apache/carbondata/api/CarbonStore.scala

```diff
@@ -145,6 +152,208 @@ object CarbonStore {
     }
   }
 
+  def showSegments(
+      limit: Option[String],
+      carbonTable: CarbonTable,
+      hadoopConf: Configuration,
+      showHistory: Boolean): Seq[Row] = {
+    val metaFolder = CarbonTablePath.getMetadataPath(carbonTable.getTablePath)
+    val loadMetadataDetailsArray = if (showHistory) {
+      SegmentStatusManager.readLoadMetadata(metaFolder) ++
+      SegmentStatusManager.readLoadHistoryMetadata(metaFolder)
+    } else {
+      SegmentStatusManager.readLoadMetadata(metaFolder)
+    }
+
+    if (loadMetadataDetailsArray.nonEmpty) {
+      var loadMetadataDetailsSortedArray = loadMetadataDetailsArray.sortWith { (l1, l2) =>
+        java.lang.Double.parseDouble(l1.getLoadName) > java.lang.Double.parseDouble(l2.getLoadName)
```

Comment:
suggest pure scala code here
```scala
l1.getLoadName.toDouble.compareTo(l2.getLoadName.toDouble)
```

Reply:
it is old code, already change to reuse code